src/System.Collections/tests/Resources/System.Collections.Tests.rd.xml

```diff
@@ -1,7 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
 <Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
-  <Library>
+  <Library Name="System.Collections.Tests">
+    <Assembly Name="*Application*" DoNotInline="true" DoNotOptimize="true" />
+    <Assembly Name="System.Collections" BinaryFormatter="All" Dynamic="Required All" DoNotInline="true" DoNotOptimize="true" />
+
     <!-- Needed because test code calls IsSerializable on these types  -->
     <Type Name="System.Collections.Generic.SortedList`2" Browse="Required Public" />
-    <Type Name="System.Collections.Generic.SortedDictionary`2" Browse="Required Public" />
+    <Type Name="System.Collections.Generic.SortedDictionary`2" Browse="Required Public">
+      <Field Name="_set" Dynamic="Required" />
```

Comment:
Does it work to use BinaryFormatter on the type instead of Dynamic on the field?